proto/regen/ecocredit/v1alpha1/query.proto

```diff
@@ -58,6 +58,51 @@ service Query {
     option (google.api.http).get =
         "/regen/ecocredit/v1alpha1/params";
   }
+
+  // SellOrder queries a sell order by its ID
+  rpc SellOrder(QuerySellOrderRequest) returns (QuerySellOrderResponse) {
+    option (google.api.http).get = "/regen/ecocredit/v1alpha1/sell-orders/id/{sell_order_id}";
+  }
+
+  // SellOrders queries a paginated list of all sell orders
+  rpc SellOrders(QuerySellOrdersRequest) returns (QuerySellOrdersResponse) {
+    option (google.api.http).get = "/regen/ecocredit/v1alpha1/sell-orders";
+  }
+
+  // SellOrdersByDenom queries a paginated list of all sell orders of a specific denom
+  rpc SellOrdersByDenom(QuerySellOrdersByDenomRequest) returns (QuerySellOrdersByDenomResponse) {
+    option (google.api.http).get = "/regen/ecocredit/v1alpha1/sell-orders/denom/{denom}";
+  }
+
+  // SellOrdersByAddress queries a paginated list of all sell orders from a specific address
+  rpc SellOrdersByAddress(QuerySellOrdersByAddressRequest) returns (QuerySellOrdersByAddressResponse) {
+    option (google.api.http).get = "/regen/ecocredit/v1alpha1/sell-orders/address/{address}";
+  }
+
+  // BuyOrder queries a buy order by its id
+  rpc BuyOrder(QueryBuyOrderRequest) returns (QueryBuyOrderResponse) {
+    option (google.api.http).get = "/regen/ecocredit/v1alpha1/buy-orders/id/{buy_order_id}";
+  }
+
+  // BuyOrders querires a paginated list of all buy orders
+  rpc BuyOrders(QueryBuyOrdersRequest) returns (QueryBuyOrdersResponse) {
+    option (google.api.http).get = "/regen/ecocredit/v1alpha1/buy-orders";
+  }
+
+  // BuyOrdersByDenom queries a paginated list of buy orders by denom
+  rpc BuyOrdersByDenom(QueryBuyOrdersByDenomRequest) returns (QueryBuyOrdersByDenomResponse) {
+    option (google.api.http).get = "/regen/ecocredit/v1alpha1/buy-orders/denom/{denom}";
+  }
```

Comment:
> would it be useful to have a query for buy orders based on the denom of the bid price?

`batch_denom` is not included in a buy order so this would be for the bid price denom. I think we can either remove this or we could make it more explicit (i.e. `bid_denom`). If we have a query for buy orders by bid price denom, we should probably have a query for sell orders by ask price denom.

Reply:
ok ill just leave it out for now, can always add it later down the road if it seems useful